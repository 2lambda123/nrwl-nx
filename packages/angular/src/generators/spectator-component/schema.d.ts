import { Schema as AngularComponentSchema } from '../component/schema';

export interface Schema extends AngularComponentSchema {
  withHost?: boolean;
  withCustomHost?: boolean;
  jest?: boolean;
}

export interface NormalizedSchema extends Schema {
  directory: string;
  filePath: string;
  projectSourceRoot: string;
  projectRoot: string;
  selector: string;
  withHost: boolean;
  withCustomHost: boolean;
  jest: boolean;
}
