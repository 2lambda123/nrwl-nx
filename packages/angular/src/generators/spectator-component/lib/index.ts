export * from './normalize-options';
export * from './validate-options';
export * from './create-files';
