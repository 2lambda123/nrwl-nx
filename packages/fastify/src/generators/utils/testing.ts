import { Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

export function createTreeWithFastifyApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  tree.write(
    'workspace.json',
    String.raw`
      {
        "projects": {
          "${appName}": {
            "root": "apps/api",
            "sourceRoot": "apps/api/src",
            "projectType": "application",
            "prefix": "api",
            "targets":{}
          }
        }
      }
    `
  );

  return tree;
}
