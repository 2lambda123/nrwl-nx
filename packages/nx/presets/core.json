{
  "implicitDependencies": {
    "package.json": {
      "dependencies": "*",
      "devDependencies": "*"
    },
    ".eslintrc.json": "*"
  },
  "tasksRunnerOptions": {
    "default": {
      "runner": "nx/tasks-runners/default",
      "options": {
        "cacheableOperations": []
      }
    }
  },
  "targetDependencies": {
    "build": [
      {
        "target": "build",
        "projects": "dependencies"
      }
    ]
  },
  "workspaceLayout": {
    "appsDir": "packages",
    "libsDir": "packages"
  }
}
