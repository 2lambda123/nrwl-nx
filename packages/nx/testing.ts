export {cold, hot} from 'jasmine-marbles';
export {readAll} from './src/testing-utils';
