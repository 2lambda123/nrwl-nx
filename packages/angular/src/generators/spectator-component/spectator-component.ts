import type { Tree } from '@nx/devkit';
import { names } from '@nx/devkit';
import { normalizeOptions, validateOptions, createFiles } from './lib';
import type { Schema } from './schema';
import { componentGenerator } from '../component/component';

export async function spectatorComponentGenerator(
  tree: Tree,
  rawOptions: Schema
) {
  validateOptions(rawOptions);
  const options = normalizeOptions(tree, rawOptions);

  const componentNames = names(options.name);
  const typeNames = names(options.type);

  await componentGenerator(tree, rawOptions);

  createFiles(tree, options, componentNames, typeNames);
}

export default spectatorComponentGenerator;
