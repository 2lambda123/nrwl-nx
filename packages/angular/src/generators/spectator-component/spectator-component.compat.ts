import { convertNxGenerator } from '@nx/devkit';
import { warnForSchematicUsage } from '../utils/warn-for-schematic-usage';
import { spectatorComponentGenerator } from './spectator-component';

export default warnForSchematicUsage(
  convertNxGenerator(spectatorComponentGenerator)
);
