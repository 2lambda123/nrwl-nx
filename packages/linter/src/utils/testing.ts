import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { Tree } from '@angular-devkit/schematics';

const migrationTestRunner = new SchematicTestRunner(
  '@nrwl/linter-migrations',
  join(__dirname, '../../migrations.json')
);

const testRunner = new SchematicTestRunner(
  '@nrwl/linter',
  join(__dirname, '../../collection.json')
);

export function runMigration<T = any>(
  migrationName: string,
  options: T,
  tree: Tree
) {
  return migrationTestRunner
    .runSchematicAsync<T>(migrationName, options, tree)
    .toPromise();
}

export function runSchematic<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}
