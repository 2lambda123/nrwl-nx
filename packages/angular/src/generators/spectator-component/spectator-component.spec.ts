import {
  addProjectConfiguration,
  stripIndents,
  updateJson,
  writeJson,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { spectatorComponentGenerator } from './spectator-component';

describe('spectator component Generator', () => {
  const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

  beforeAll(async () => {
    // ARRANGE
    addProjectConfiguration(tree, 'lib1', {
      projectType: 'library',
      sourceRoot: 'libs/lib1/src',
      root: 'libs/lib1',
    });
    tree.write(
      'libs/lib1/src/lib/lib.module.ts',
      `
     import { NgModule } from '@angular/core';
     
     @NgModule({
       declarations: [],
       exports: []
     })
     export class LibModule {}`
    );
    tree.write('libs/lib1/src/index.ts', 'export * from "./lib/lib.module";');
  });

  it('should create component files correctly', async () => {
    // ACT
    await spectatorComponentGenerator(tree, {
      name: 'example',
      project: 'lib1',
    });

    // ASSERT
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.ts', 'utf-8')
    ).toMatchSnapshot('component');
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.html', 'utf-8')
    ).toMatchSnapshot('template');
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.css', 'utf-8')
    ).toMatchSnapshot('stylesheet');
    expect(
      tree.read('libs/lib1/src/lib/example/example.component.spec.ts', 'utf-8')
    ).toMatchSnapshot('component test file');
    expect(tree.read('libs/lib1/src/index.ts', 'utf-8')).toMatchSnapshot(
      'entry point file'
    );
  });

  it('should not generate test file when --skip-tests=true', async () => {
    // ACT
    await spectatorComponentGenerator(tree, {
      name: 'example',
      project: 'lib1',
      skipTests: true,
    });

    // ASSERT
    expect(
      tree.exists('libs/lib1/src/lib/example/example.component.spec.ts')
    ).toBe(false);
  });

  describe('component', () => {
    it('should generate tests using jest', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        jest: true,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).toContain('jest');
    });

    it('should not generate tests using jest', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        jest: false,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).not.toContain('jest');
    });
  });

  describe('componentCustomHost', () => {
    it('should generate tests using jest', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        withCustomHost: true,
        jest: true,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).toContain('jest');
    });

    it('should not generate tests using jest', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        withCustomHost: true,
        jest: false,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).not.toContain('jest');
    });

    it('should not generate tests using host', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        withCustomHost: true,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).toContain('CustomHostComponent');
    });
  });

  describe('componentHost', () => {
    it('should generate tests using jest', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        withHost: true,
        jest: true,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).toContain('jest');
    });

    it('should not generate tests using jest', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        withHost: true,
        jest: false,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).not.toContain('jest');
    });

    it('should not generate tests using host', async () => {
      // ACT
      await spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        withHost: true,
      });

      // ASSERT
      expect(
        tree.read(
          'libs/lib1/src/lib/example/example.component.spec.ts',
          'utf-8'
        )
      ).toContain('createHostFactory');
    });
  });

  it('should throw error when both withHost, withCustomHost are provided', async () => {
    // ACT & ASSERT
    await expect(
      spectatorComponentGenerator(tree, {
        name: 'example',
        project: 'lib1',
        withHost: true,
        withCustomHost: true,
      })
    ).rejects.toThrow(
      'The provided options are invalid. Please provide either withCustomHost or withHost.'
    );
  });
});
