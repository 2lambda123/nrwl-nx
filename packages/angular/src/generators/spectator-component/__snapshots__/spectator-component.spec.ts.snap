// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spectator component Generator should create component files correctly: component 1`] = `
"import { Component } from '@angular/core';

@Component({
  selector: 'proj-example',
  templateUrl: './example.component.html',
  styleUrls: ['./example.component.css'],
})
export class ExampleComponent {}
"
`;

exports[`spectator component Generator should create component files correctly: component test file 1`] = `
"import { Spectator, createComponentFactory } from '@ngneat/spectator';

import { ExampleComponent } from './example.component';

describe('ExampleComponent', () => {
  let spectator: Spectator<ExampleComponent>;
  const createComponent = createComponentFactory(ExampleComponent);

  it('should create', () => {
    spectator = createComponent();

    expect(spectator.component).toBeTruthy();
  });
});
"
`;

exports[`spectator component Generator should create component files correctly: entry point file 1`] = `
"export * from './lib/lib.module';
"
`;

exports[`spectator component Generator should create component files correctly: stylesheet 1`] = `""`;

exports[`spectator component Generator should create component files correctly: template 1`] = `
"<p>example works!</p>
"
`;
